exporter/metric/metric.go

```diff
@@ -627,5 +628,21 @@ func numberDataPointToValue[N int64 | float64](
 // > * Label name must start with a letter or digit.
 // > * The maximum length of a label name is 100 characters.
 func normalizeLabelKey(s string) string {
-	return strings.ReplaceAll(s, ".", "_")
+	if len(s) == 0 {
+		return s
+	}
+	s = strings.Map(sanitizeRune, s)
+	if unicode.IsDigit(rune(s[0])) {
```

Comment:
the comments above say `Label name must start with a letter or digit`, is that not accurate?

Reply:
there's also no check on the length, do we just truncate?